File: skills/astra-toolkit/clients/python/examples/tables-alter-type-add.py
import os
from astrapy import DataAPIClient
from astrapy.info import (
    AlterTypeAddFields,
    ColumnType,
    TableScalarColumnTypeDescriptor,
)

# Get a database
client = DataAPIClient()
database = client.get_database(
    os.environ["ASTRA_DB_API_ENDPOINT"], token=os.environ["ASTRA_DB_APPLICATION_TOKEN"]
)

# Add fields to a user-defined type
database.alter_type(
    "member",
    AlterTypeAddFields(
        fields={
            "email": TableScalarColumnTypeDescriptor(ColumnType.TEXT),
            "credits": TableScalarColumnTypeDescriptor(ColumnType.INT),
        }
    ),
)

File: skills/astra-toolkit/clients/java/examples/tables-alter-type-add.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.commands.AlterTypeAddFields;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient(System.getenv("ASTRA_DB_APPLICATION_TOKEN")).getDatabase(System.getenv("ASTRA_DB_API_ENDPOINT"));

    // Add fields to a user-defined type
    database.alterType(
        "member", new AlterTypeAddFields().addFieldText("email").addFieldInt("credits"));
  }
}
